pkg/codegen/go/gen.go

```diff
@@ -311,13 +311,33 @@ func printComment(w io.Writer, comment string, indent bool) {
 }
 
 func genInputInterface(w io.Writer, name string) {
+	printComment(w, getInputUsage(name), false)
 	fmt.Fprintf(w, "type %sInput interface {\n", name)
 	fmt.Fprintf(w, "\tpulumi.Input\n\n")
 	fmt.Fprintf(w, "\tTo%sOutput() %sOutput\n", title(name), name)
 	fmt.Fprintf(w, "\tTo%sOutputWithContext(context.Context) %sOutput\n", title(name), name)
 	fmt.Fprintf(w, "}\n\n")
 }
 
+func getInputUsage(name string) string {
```

Comment:
Do we want some more descriptive text at the beginning of all of these ?

For the builtins we have a comment like:

```go
// StringPtrInput is an input type that accepts StringPtr and StringPtrOutput values.
```

Should we here have:

```go
// FooPtrInput is an input type that accepts FooArgs, FooPtr and FooPtrOutput values.
// You can construct a concrete instance of `FooPtrInput` with either:
//
//    FooArgs{...}
//
// or:
// 
//    nil
```

Reply:
Note that we'll also need to begin these comments with `FooPtrInput` or we won't be lint-clean.

Reply:
Good call, updated